src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/updatelistingrequirements/UpdateListingRequirementsHandler.java

```diff
@@ -0,0 +1,210 @@
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.tika.utils.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DynamicList;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DynamicListItem;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingInterpreter;
+import uk.gov.hmcts.reform.sscs.ccd.domain.OverrideFields;
+import uk.gov.hmcts.reform.sscs.ccd.domain.ReservedToMember;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.sscs.client.JudicialRefDataApi;
+import uk.gov.hmcts.reform.sscs.idam.IdamService;
+import uk.gov.hmcts.reform.sscs.model.client.JudicialMemberAppointments;
+import uk.gov.hmcts.reform.sscs.model.client.JudicialRefDataUsersRequest;
+import uk.gov.hmcts.reform.sscs.model.client.JudicialRefDataUsersResponse;
+import uk.gov.hmcts.reform.sscs.model.client.JudicialUser;
+import uk.gov.hmcts.reform.sscs.reference.data.model.JudicialMemberType;
+import uk.gov.hmcts.reform.sscs.reference.data.model.Language;
+import uk.gov.hmcts.reform.sscs.reference.data.service.SignLanguagesService;
+import uk.gov.hmcts.reform.sscs.reference.data.service.VerbalLanguagesService;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class UpdateListingRequirementsHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    @Value("${feature.snl.enabled}")
+    private boolean isScheduleListingEnabled;
+
+    private final IdamService idamService;
+    private final SignLanguagesService signLanguagesService;
+    private final VerbalLanguagesService verbalLanguagesService;
+    private final JudicialRefDataApi judicialRefData;
+
+    public static final String SERVICE_NAME = "sscs";
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        final SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+
+        if (isScheduleListingEnabled) {
+            OverrideFields overrideFields = sscsCaseData.getSchedulingAndListingFields().getOverrideFields();
+
+            if (isNull(overrideFields)) {

```

Comment:
You could move this into the getter on SscsCaseData (see scheduling and listing fields)